src/transaction/mvcc_table.cpp

```diff
@@ -311,6 +311,12 @@ mvcctable::build_mvcc_info (log_tdes &tdes)
 	  // no version change; copying status was successful
 	  break;
 	}
+      else
+	{
+	  // a failed copy may break data validity; to make sure next copy is not affected, it is better to reset
+	  // bit area.
+	  tdes.mvccinfo.snapshot.m_active_mvccs.reset_bit_area ();
```

Comment:
How about reset all fields?

Reply:
resetting bit_area was enough to fix the issue; for clarity of purpose, I renamed the function as reset_active_transactions and also set m_long_tran_mvccids_length to 0.